gallery_dl/extractor/motherless.py

```diff
@@ -0,0 +1,308 @@
+            "count": get_gallery_image_count(page) + get_gallery_video_count(page)}
+        
+        yield Message.Directory, data
+
+        for id, url, extension, title, num in get_images(self):
+            data |= {
+                "id": id,
+                "filename": id,
+                "extension": extension,
+                "title": title,
+                "num": num}
+
+            yield Message.Url, url, data
+
+        for id, url, title, num in get_videos(self):
+            data |= {
+                "id": id,
+                "filename": id,
+                "extension": "mp4",
+                "title": title,
+                "num": num}
+
+            yield Message.Url, url, data
+
+
+# Url extractors.
+
+def get_images(extractor):
+    n = 1
+    total_count = 0
+
+    while True:
+        page = extractor.request(f"{extractor.root}/GI{extractor.gallery_id}?page={n}").text
+        page_count = 0
+
+        for result in re.finditer(f' src="https:\/\/cdn5-thumbs\.motherlessmedia\.com\/thumbs\/([A-Z0-9]+?)\.(jpg|gif)"[\s\S]+?alt="(.+)"', page):
+            id = result.group(1)
+            url = f"https://cdn5-images.motherlessmedia.com/images/{id}.jpg"
+            extension = result.group(2)
+            title = result.group(3)
+            page_count += 1
+
+            yield id, url, extension, title, total_count + page_count
+
+        if page_count == 0:
+            return
+
+        total_count += page_count
+        n += 1
+
+def get_videos(extractor):
+    n = 1
+    total_count = 0
+
+    while True:
+        page = extractor.request(f"{extractor.root}/GV{extractor.gallery_id}?page={n}").text
+        page_count = 0
+
+        for result in re.finditer(f'thumbs\/([A-Z0-9]+?)-strip\.jpg" alt="(.+)"', page):
+            id = result.group(1)
+            url = f"https://cdn5-videos.motherlessmedia.com/videos/{id}.mp4"
+            title = result.group(2)
+            page_count += 1
+
+            yield id, url, title, total_count + page_count
+
+        if page_count == 0:
+            return
+
+        total_count += page_count
+        n += 1
+
+# Metadata extractors.
+
+def get_media_tags(page_data):
+    try:
+        tags_html = re.search('<div class="media-meta-tags">([\S\s]+?)</div>', page_data).group(1)
+    except AttributeError:
+        # No tags found.
+        return []
+
+    tags = text.split_html(tags_html)
+    for i, tag in enumerate(tags):
+        tags[i] = tag.replace('#', '')
+
+    return tags
+
+def get_media_title(page_data):
+    title = re.search('<div class="media-meta-title">([\S\s]+?)</div>', page_data).group(1)
+    return unescape(text.remove_html(title))
+
+def get_media_date(page_data):
+    # Find date uploaded and convert to ISO 8601.
+    try:
+        # Find 'DD Mon YYYY' format.
+        date = re.search('<span class="count">(\d{1,2}\s+\w+\s+\d{4})</span>', page_data).group(1)
+        return text.parse_datetime(date, "%d  %b  %Y").isoformat()
```

Comment:
no need for double spaces `%d  %b  %Y` => `%d %b %Y`

Reply:
Done